runtime/client_http_request.go

```diff
@@ -129,8 +113,37 @@ func (req *ClientHTTPRequest) WriteJSON(
 // Do will send the request out.
 func (req *ClientHTTPRequest) Do(
 	ctx context.Context,
-) (*ClientHTTPResponse, error) {
-	res, err := req.client.Client.Do(
+) (resp *ClientHTTPResponse, err error) {
+	defer func() {
+		scope := req.client.scope.Tagged(map[string]string{
```

Comment:
It makes more sense to subscope the method scope at where the method is defined, not in the lower layer where request is made.

Reply:
Creating pre-allocated metrics where the method created.